docs/architecture/adr-013-multisigs.md

```diff
@@ -0,0 +1,148 @@
+```
+The multisignature will be converted to an `[]byte` using amino.MarshalBinaryBare. \*
+
+#### Bit Array
+We would be using a new implementation of a bitarray. The struct it would be encoded/decoded from is 
+```golang
+type CryptoBitArray struct {
+	ExtraBitsStored  byte      `json:"extra_bits"` // The number of extra bits in elems. 
+	Elems            []byte    `json:"elems"`
+}
+```
+The reason for not using the BitArray currently implemented in `libs/common/bit_array.go`
+is that it is less space efficient, due to a space / time trade-off.
+Evidence for this is outlined in [this issue](https://github.com/tendermint/tendermint/issues/2077). 
+
+In the multisig, we will not be performing arithmetic operations,
+so there is no performance increase with the current implementation,
+and just loss of spatial efficiency.
+Implementing this new bit array with `[]byte` _should_ be simple, as no
+arithmetic operations between bit arrays are required, and save a couple of bytes.
+(Explained in that same issue)
+
+When this bit array encoded, the number of elements is encoded due to amino.
+However we may be encoding a full byte for what we actually only need 1-7 bits for.
+We store that difference in ExtraBitsStored.
+This allows for us to have an unbounded number of signers, and is more space efficient than what is currently used in `libs/common`.
+Again the implementation of this space saving feature is straight forward.
+
+### Encoding the structs
+
+We will use straight forward amino encoding. This is chosen for ease of compatibility in other languages. 
+
+#### Footnotes
+\* I advocate that we should actually include the fields from CryptoBitArray into the multisignature directly. This saves space in encoding, despite breaking abstractions slightly. I think it is worth the space savings, however I've put this into a footnote since this is quite minor. 
+
+### Future points of discussion
+
+If desired, we can use ed25519 batch verification for all ed25519 keys.
+This is a future point of discussion, but would be backwards compatible as this information won't need to be marshalled.
+(There may even be cofactor concerns without ristretto)
+Aggregation of pubkeys / sigs in Schnorr sigs / BLS sigs is not backwards compatible, and would need to be a new ASM type.
+
+## Status
+
+Proposed. Needs a decision on the item mentioned in the footnote, and the ADR as a whole. 
+
+## Consequences
+
+### Positive
+* Supports multisignatures, in a way that won't require any special cases in our downstream verification code. 
+* Easy to serialize / deserialize
+* Unbounded number of signers
+
+### Negative
+* Larger codebase, however this should reside in a subfolder of tendermint/crypto, as it provides no new interfaces. (Ref #https://github.com/tendermint/go-crypto/issues/136)
+* Space inefficient due to utilization of amino encoding
+* Suggested implementation requires a new struct for every ASM. 
```

Comment:
A new struct for every ASM ?

Reply:
This is required since we're registering another concrete type with amino for each ASM we want to support.